backend/app/services/api_manager.py

```
@@ -0,0 +1,627 @@
+            return False
+
+    async def clear_pattern(self, pattern: str) -> int:
+        """Clear cache entries matching pattern."""
+        if not self._connected:
+            return 0
+
+        try:
+            keys = await self._redis.keys(pattern)
+            if keys:
+                return await self._redis.delete(*keys)
+            return 0
```

Ruff (0.13.3)

Consider moving this statement to an `else` block

(TRY300)